File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    // 정답을 담을 answer 배열
    var answer = [0, 0];
    
    // sizes 배열의 크기
    const n = sizes.length;
    
    // sizes를 순회하며 w, h를 갱신할지 비교
    for (let i = 0; i < n; i++) {
        // 현재 명함과 정답 배열 정렬
        sizes[i].sort((a, b) => a - b);
        console.log(sizes[i])
        
        // answer의 w, h로 현재 명합을 담을 수 없는 경우
        // - answer의 w, h 중 더 긴 길이와 현재 명함의 w, h 중 더 긴 길이를 비교
        // - answer보다 현재 명함이 더 길면 갱신
        if (answer[1] < sizes[i][1]) {
            answer[1] = sizes[i][1];
        }
        if (answer[0] < sizes[i][0]) {
            answer[0] = sizes[i][0];
        }
    }
    
    return answer[0] * answer[1];
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int min = Integer.MIN_VALUE;
        int max = Integer.MIN_VALUE;
        
        for (int[] size :sizes) {
            if (size[0] < size[1]) {
                min = Math.max(min, size[0]);
                max = Math.max(max, size[1]);
            } else {
                min = Math.max(min, size[1]);
                max = Math.max(max, size[0]);
            }
        }
        
        return min * max;
    }
}